### ISSUE TITLE
The 'old' login page is going to be retired in the near future

### ISSUE THREAD

**Issue by rsudev** [cgeo/cgeo#6404]
We got a mail from Bryan today:

> Please note that in the near future, our team will deprecate the [old log-in page](https://www.geocaching.com/login/default.aspx) and route folks to the [new log-in page](https://www.geocaching.com/account/login?returnUrl=%2fplay). We're doing this so that we're not maintaining multiple log-in pages.

We need to update:

- [x] release (by merging the fix branch)
- [x] master (by merging the fix branch)
- [x] cgeo-legacy (by cherry-picking from `release`)

**Comment by Lineflyer** [comment]
Can someone find out if the new login page will break the login?
That might force us to release soon...but luckily we are almost ready to release.

**Comment by rsudev** [comment]
The new login page exists already quite a while, but we still use the old one in code. Switching shouldn't be a big deal, but needs to be done.

**Comment by Lineflyer** [comment]
Ideally supporting both would be wise as we don't know the exact point they will change it (and might revert and come back and revert ;) )

**Comment by samueltardieu** [comment]
There is no reason for them to revert, as @rsudev said it has been existing for a while.

**Comment by Lineflyer** [comment]
Ok if it exists and is working we can start using it asap I asume.

**Comment by Lineflyer** [comment]
BTW: I will answer Bryan tomorrow.

**Comment by Lineflyer** [comment]
Is someone looking into this already?

**Comment by Lineflyer** [comment]
On my PC I am already directed to the new login page, however the old one still exists right now. But it seems they are already live with that change.

**Comment by mucek4** [comment]
Maybe not related to this login issue. But do they have any "staging" environment where we can try this new releases?

**Comment by kumy** [comment]
http://staging.geocaching.com/ ?

**Comment by Lineflyer** [comment]
Yes, that their staging website. But as we do not know status of the code used there, it makes no sense to just use it for testing anything.
However if they continue informing us about upcoming important changes, they might also link us to there if it is useful. Right now for the login issue it is not required as the new login page is already in production.

**Comment by mucek4** [comment]
That is true. But if we run tests on staging env. Or build a c:geo version that uses stanging, we might be able to test their changes before they hit production?

**Comment by Lineflyer** [comment]
Make no sense to me as we don't know whats happening on this environment (e.g. it might also be a sandbox to play around, so in case we find a problem there we still don't know what to conclude out of that).
Furthermore I would not do that, without prior asking them anyway. But instead of doing that, I would IMHO instead put effort into an API evolvement discussion.

**Comment by Lineflyer** [comment]
Is someone working on this?
We are almost ready for release, just a fix for this issue is pending.

**Comment by samueltardieu** [comment]
Merged in `master` for testing, waiting for @Lineflyer output on the next nightly.

**Comment by rsudev** [comment]
Short feedback from a smoke-test on master: works for me! Thanks @samueltardieu !

**Comment by Lineflyer** [comment]
Also working for me in a smoke test. Will test more tonight.
@rsudev Is it also taking so long for you from app start until Login OK is shown on mainscreen? It takes >30 seconds for me.

**Comment by rsudev** [comment]
@Lineflyer: No, it took ~10 seconds right now (which is in the same order of magnitude as before). Server lag?

**Comment by Lineflyer** [comment]
I got a bit deeper and looked into the logs. Before starting the login c:geo has been closed and killed.
Both scenarios have been tested several times with same result.

This is what I see:

### Old login process
This step take 1,7 seconds :
20:43:24.708 1st GET request to login page
20:43:26.446 Info: Already logged in

This step takes 2,6 seconds:
20:43:26.446 Other login steps (location, date format, etc.)
20:43:29.054 End of login process

**Total duration: 4,3 seconds**

### New login process
This step takes 2 seconds:
03-16 20:51:24.558  1st GET request to login page
03-16 20:51:25.399  2nd GET request to login page
03-16 20:51:26.577  1st POST request to login page

This step takes 6,2 seconds
03-16 20:51:28.610  1st GET request to souvenirs page 
03-16 20:51:32.311  1st POST request to souvenirs page
03-16 20:51:34.845  Info: Language changed to English

This step takes 3,5 seconds:
03-16 20:51:34.848  3rd GET request to login page
03-16 20:51:38.363  Info: Already logged in

This step takes 18,2 seconds:
03-16 20:51:38.368  2nd GET request to souvenirs page 
03-16 20:51:47.883  2nd POST request to souvenirs page
03-16 20:51:56.499  Info: Language changed to English

This step takes 2 seconds:
03-16 20:51:56.504  Other login steps (location, date format, etc.)
03-16 20:51:58.589  End of login process

**Total duration: 31,9 seconds**


### Conclusions:
- With the old login process I am recognized as logged in on the first try. With the new login a relogin always occurs. Is this a Cookie problem?
- With the new login process the Switch to English is triggered although the page should already be in English. Furthermore the switch to english is triggered two times and again followed by a login (in which I am now remembered as being logged in). 
- Reason for the big time delay is, that the switch to english page (souvenirs.aspx) is long to retrieve from server. So the time delay is not the main problem here, but the reason why switch to english is triggered at all.

Old login adb log:
```
20:43:24.708 Debug cgeo 25607  [OkHttp] GET https://status.cgeo.org/api/status.json?version_code=20170314&version_name=2017.03.14-f4bba00%20developer%20build&locale=de_DE
20:43:24.708 Info cgeo 25607  [main] Starting cgeo.geocaching 20170314 a.k.a 2017.03.14-f4bba00 developer build
20:43:24.708 Debug cgeo 25607  [RxCachedThreadScheduler-2] Database clean: started
20:43:24.708 Debug cgeo 25607  [RxCachedThreadScheduler-2] Database clean: removing 0 geocaches
20:43:24.708 Debug cgeo 25607  [RxCachedThreadScheduler-2] Database clean: removing non-existing lists
20:43:24.708 Debug cgeo 25607  [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from attributes
20:43:24.708 Debug cgeo 25607  [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from spoilers
20:43:24.708 Debug cgeo 25607  [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from lists
20:43:24.708 Debug cgeo 25607  [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from waypoints
20:43:24.708 Debug cgeo 25607  [OkHttp] GET https://www.geocaching.com/login/default.aspx
20:43:24.708 Debug cgeo 25607  [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from trackables
20:43:24.708 Debug cgeo 25607  [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from logcount
20:43:24.708 Debug cgeo 25607  [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from logs offline
20:43:24.816 Debug cgeo 25607  [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from logs
20:43:24.816 Debug cgeo 25607  [RxCachedThreadScheduler-2] Database clean: removing non-existing logs from logimages
20:43:24.816 Debug cgeo 25607  [RxCachedThreadScheduler-2] Database clean: finished
20:43:25.034 Debug cgeo 25607  [main] LocationProvider: requesting most precise locations
20:43:25.468 Debug cgeo 25607  [OkHttp] 200 (809 ms) GET https://status.cgeo.org/api/status.json?version_code=20170314&version_name=2017.03.14-f4bba00%20developer%20build&locale=de_DE (http/1.1)
20:43:26.120 Debug cgeo 25607  [OkHttp] 200 (1400 ms) GET https://www.geocaching.com/login/default.aspx (http/1.1)
20:43:26.446 Info cgeo 25607  [network--2] Already logged in Geocaching.com as Lineflyer (PREMIUM)
20:43:26.446 Info cgeo 25607  [network--2] Geocaching.com language already set to English
20:43:26.446 Debug cgeo 25607  [OkHttp] GET https://www.geocaching.com/email/
20:43:26.772 Debug cgeo 25607  [OkHttp] 200 (331 ms) GET https://www.geocaching.com/email/ (http/1.1)
20:43:27.207 Debug cgeo 25607  [OkHttp] GET https://www.geocaching.com/account/settings/homelocation
20:43:27.207 Debug cgeo 25607  [OkHttp] GET https://www.geocaching.com/account/settings/membership
20:43:27.207 Debug cgeo 25607  [OkHttp] GET https://www.geocaching.com/account/settings/preferences
20:43:27.642 Debug cgeo 25607  [OkHttp] 200 (514 ms) GET https://www.geocaching.com/account/settings/homelocation (http/1.1)
20:43:28.403 Debug cgeo 25607  [OkHttp] 200 (1217 ms) GET https://www.geocaching.com/account/settings/membership (http/1.1)
20:43:28.620 Debug cgeo 25607  [OkHttp] 200 (1384 ms) GET https://www.geocaching.com/account/settings/preferences (http/1.1)
20:43:28.946 Debug cgeo 25607  [OkHttp] Setting member status to PREMIUM
20:43:29.054 Debug cgeo 25607  [network--2] Setting GC custom date to dd MMM yy
```

New login adb log:
```
03-16 20:51:24.505 11615 11636 D cgeo    : [OkHttp] GET https://status.cgeo.org/api/status.json?version_code=30170316&version_name=2017.03.16-NB1-27b9c16&locale=de_DE&installer=com.google.android.packageinstaller
03-16 20:51:24.538 11615 11615 I cgeo    : [main] Starting cgeo.geocaching 30170316 a.k.a 2017.03.16-NB1-27b9c16
03-16 20:51:24.540 11615 11640 D cgeo    : [RxCachedThreadScheduler-2] Database clean: started
03-16 20:51:24.546 11615 11640 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing 0 geocaches
03-16 20:51:24.547 11615 11640 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing lists
03-16 20:51:24.547 11615 11640 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from attributes
03-16 20:51:24.548 11615 11640 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from spoilers
03-16 20:51:24.549 11615 11640 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from lists
03-16 20:51:24.550 11615 11640 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from waypoints
03-16 20:51:24.551 11615 11640 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from trackables
03-16 20:51:24.552 11615 11640 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from logcount
03-16 20:51:24.553 11615 11640 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from logs offline
03-16 20:51:24.553 11615 11640 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from logs
03-16 20:51:24.558 11615 11644 D cgeo    : [OkHttp] GET https://www.geocaching.com/account/login?ReturnUrl=/play
03-16 20:51:24.572 11615 11640 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing logs from logimages
03-16 20:51:24.579 11615 11640 D cgeo    : [RxCachedThreadScheduler-2] Database clean: finished
03-16 20:51:24.738 11615 11615 D cgeo    : [main] LocationProvider: requesting most precise locations
03-16 20:51:25.329 11615 11644 D cgeo    : [OkHttp] 200 (771 ms) GET https://www.geocaching.com/account/login?ReturnUrl=/play (http/1.1)
03-16 20:51:25.399 11615 11644 D cgeo    : [OkHttp] GET https://www.geocaching.com/account/login?ReturnUrl=/play
03-16 20:51:26.007 11615 11636 D cgeo    : [OkHttp] 200 (1502 ms) GET https://status.cgeo.org/api/status.json?version_code=30170316&version_name=2017.03.16-NB1-27b9c16&locale=de_DE&installer=com.google.android.packageinstaller (http/1.1)
03-16 20:51:26.301 11615 11644 D cgeo    : [OkHttp] 200 (902 ms) GET https://www.geocaching.com/account/login?ReturnUrl=/play (http/1.1)
03-16 20:51:26.577 11615 11644 D cgeo    : [OkHttp] POST https://www.geocaching.com/account/login?ReturnUrl=/play
03-16 20:51:27.701 11615 11644 D cgeo    : [OkHttp] 200 (1123 ms) POST https://www.geocaching.com/account/login?ReturnUrl=/play (http/1.1) (=> https://www.geocaching.com/play/search)
03-16 20:51:28.610 11615 11644 D cgeo    : [OkHttp] GET https://www.geocaching.com/my/souvenirs.aspx
03-16 20:51:30.539 11615 11644 D cgeo    : [OkHttp] 200 (1929 ms) GET https://www.geocaching.com/my/souvenirs.aspx (http/1.1)
03-16 20:51:32.311 11615 11644 D cgeo    : [OkHttp] POST https://www.geocaching.com/my/souvenirs.aspx
03-16 20:51:34.844 11615 11644 D cgeo    : [OkHttp] 200 (2533 ms) POST https://www.geocaching.com/my/souvenirs.aspx (http/1.1)
03-16 20:51:34.845 11615 11643 I cgeo    : [network--2] changed language on geocaching.com to English
03-16 20:51:34.848 11615 11644 D cgeo    : [OkHttp] GET https://www.geocaching.com/account/login?ReturnUrl=/play
03-16 20:51:36.254 11615 11644 D cgeo    : [OkHttp] 200 (1405 ms) GET https://www.geocaching.com/account/login?ReturnUrl=/play (http/1.1) (=> https://www.geocaching.com/play/search)
03-16 20:51:38.363 11615 11643 I cgeo    : [network--2] Already logged in Geocaching.com as Lineflyer (PREMIUM)
03-16 20:51:38.368 11615 11644 D cgeo    : [OkHttp] GET https://www.geocaching.com/my/souvenirs.aspx
03-16 20:51:38.812 11615 11644 D cgeo    : [OkHttp] 200 (444 ms) GET https://www.geocaching.com/my/souvenirs.aspx (http/1.1)
03-16 20:51:47.883 11615 11644 D cgeo    : [OkHttp] POST https://www.geocaching.com/my/souvenirs.aspx
03-16 20:51:56.498 11615 11644 D cgeo    : [OkHttp] 200 (8614 ms) POST https://www.geocaching.com/my/souvenirs.aspx (http/1.1)
03-16 20:51:56.499 11615 11643 I cgeo    : [network--2] changed language on geocaching.com to English
03-16 20:51:56.504 11615 11636 D cgeo    : [OkHttp] GET https://www.geocaching.com/account/settings/membership
03-16 20:51:56.504 11615 11644 D cgeo    : [OkHttp] GET https://www.geocaching.com/account/settings/homelocation
03-16 20:51:56.506 11615 11691 D cgeo    : [OkHttp] GET https://www.geocaching.com/account/settings/preferences
03-16 20:51:57.451 11615 11636 D cgeo    : [OkHttp] 200 (947 ms) GET https://www.geocaching.com/account/settings/membership (http/1.1)
03-16 20:51:57.451 11615 11644 D cgeo    : [OkHttp] 200 (947 ms) GET https://www.geocaching.com/account/settings/homelocation (http/1.1)
03-16 20:51:57.680 11615 11691 D cgeo    : [OkHttp] 200 (1174 ms) GET https://www.geocaching.com/account/settings/preferences (http/1.1)
03-16 20:51:57.973 11615 11643 D cgeo    : [network--2] Setting GC custom date to dd MMM yy
03-16 20:51:58.589 11615 11636 D cgeo    : [OkHttp] Setting member status to PREMIUM
```

**Comment by Lineflyer** [comment]
From looking at the code it could IMHO be that `GCConstants.PATTERN_MAP_LOGGED_IN` need to be changed to detect "Already logged in state" and as such avoiding going into `switchToEnglish()` (which we for sure should always do first if not already logged in).

What I do not yet understand is why `switchToEnglish()` is triggered two times during the whole process.

**Comment by rsudev** [comment]
Perhaps it is generally not a good idea to use the souvenirs page as all. IIRC we chose it, because it was the smallest one. But I guess for avid cachers it will be 'expensive' to be delivered by the backend servers and therefor take more time than we want it to take.

**Comment by Lineflyer** [comment]
Well yes, the sourvenirs page is not the fastest one as they seem to collect the sourvenir state online from their database or alike. We choose it AFAIR as this is most unlikely to change.

However in normal cases switch to english should only be rarely executed at all.

**Comment by samueltardieu** [comment]
@Lineflyer I am building -NB1 with changes to reduce the number of requests, please test.

**Comment by Lineflyer** [comment]
@samueltardieu Thanks! Will do.

**Comment by Lineflyer** [comment]
Quick feedback (will test some failure cases and switch to english later):
Looks good now for the standard case.

```
03-17 17:17:52.664 23004 23025 D cgeo    : [OkHttp] GET https://status.cgeo.org/api/status.json?version_code=30170317&version_name=2017.03.17-NB1-07f2df9&locale=de_DE&installer=com.google.android.packageinstaller
03-17 17:17:52.692 23004 23004 I cgeo    : [main] Starting cgeo.geocaching 30170317 a.k.a 2017.03.17-NB1-07f2df9
03-17 17:17:52.695 23004 23029 D cgeo    : [RxCachedThreadScheduler-2] Database clean: started
03-17 17:17:52.723 23004 23033 D cgeo    : [OkHttp] GET https://www.geocaching.com/account/login
03-17 17:17:52.766 23004 23029 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing 0 geocaches
03-17 17:17:52.769 23004 23029 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing lists
03-17 17:17:52.770 23004 23029 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from attributes
03-17 17:17:52.777 23004 23029 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from spoilers
03-17 17:17:52.781 23004 23029 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from lists
03-17 17:17:52.782 23004 23029 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from waypoints
03-17 17:17:52.789 23004 23029 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from trackables
03-17 17:17:52.804 23004 23029 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from logcount
03-17 17:17:52.809 23004 23029 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from logs offline
03-17 17:17:52.809 23004 23029 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from logs
03-17 17:17:52.968 23004 23004 D cgeo    : [main] LocationProvider: requesting most precise locations
03-17 17:17:53.031 23004 23029 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing logs from logimages
03-17 17:17:53.053 23004 23029 D cgeo    : [RxCachedThreadScheduler-2] Database clean: finished
03-17 17:17:53.233 23004 23025 D cgeo    : [OkHttp] 200 (569 ms) GET https://status.cgeo.org/api/status.json?version_code=30170317&version_name=2017.03.17-NB1-07f2df9&locale=de_DE&installer=com.google.android.packageinstaller (http/1.1)
03-17 17:17:53.811 23004 23033 D cgeo    : [OkHttp] 200 (1088 ms) GET https://www.geocaching.com/account/login (http/1.1) (=> https://www.geocaching.com/play/search)
03-17 17:17:54.144 23004 23032 I cgeo    : [network--2] Already logged in Geocaching.com as Lineflyer (PREMIUM)
03-17 17:17:54.519 23004 23032 I cgeo    : [network--2] Geocaching.com language already set to English
03-17 17:17:54.520 23004 23033 D cgeo    : [OkHttp] GET https://www.geocaching.com/email/
03-17 17:17:54.732 23004 23033 D cgeo    : [OkHttp] 200 (212 ms) GET https://www.geocaching.com/email/ (http/1.1)
03-17 17:17:54.748 23004 23033 D cgeo    : [OkHttp] GET https://www.geocaching.com/account/settings/homelocation
03-17 17:17:54.749 23004 23025 D cgeo    : [OkHttp] GET https://www.geocaching.com/account/settings/membership
03-17 17:17:54.754 23004 23065 D cgeo    : [OkHttp] GET https://www.geocaching.com/account/settings/preferences
03-17 17:17:54.960 23004 23033 D cgeo    : [OkHttp] 200 (212 ms) GET https://www.geocaching.com/account/settings/homelocation (http/1.1)
03-17 17:17:55.444 23004 23025 D cgeo    : [OkHttp] 200 (695 ms) GET https://www.geocaching.com/account/settings/membership (http/1.1)
03-17 17:17:55.617 23004 23065 D cgeo    : [OkHttp] 200 (863 ms) GET https://www.geocaching.com/account/settings/preferences (http/1.1)
03-17 17:17:55.668 23004 23025 D cgeo    : [OkHttp] Setting member status to PREMIUM
03-17 17:17:55.853 23004 23032 D cgeo    : [network--2] Setting GC custom date to dd MMM yy
```

**Comment by Lineflyer** [comment]
Switch to english working normal (Scenario: User switched website to german before starting c:geo):

```
03-17 17:24:04.856 25082 25106 D cgeo    : [OkHttp] GET https://status.cgeo.org/api/status.json?version_code=30170317&version_name=2017.03.17-NB1-07f2df9&locale=de_DE&installer=com.google.android.packageinstaller
03-17 17:24:04.894 25082 25082 I cgeo    : [main] Starting cgeo.geocaching 30170317 a.k.a 2017.03.17-NB1-07f2df9
03-17 17:24:04.896 25082 25111 D cgeo    : [RxCachedThreadScheduler-2] Database clean: started
03-17 17:24:04.920 25082 25115 D cgeo    : [OkHttp] GET https://www.geocaching.com/account/login
03-17 17:24:04.948 25082 25111 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing 0 geocaches
03-17 17:24:04.951 25082 25111 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing lists
03-17 17:24:04.952 25082 25111 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from attributes
03-17 17:24:04.956 25082 25111 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from spoilers
03-17 17:24:04.958 25082 25111 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from lists
03-17 17:24:04.959 25082 25111 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from waypoints
03-17 17:24:04.963 25082 25111 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from trackables
03-17 17:24:04.973 25082 25111 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from logcount
03-17 17:24:04.978 25082 25111 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from logs offline
03-17 17:24:04.979 25082 25111 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing caches from logs
03-17 17:24:05.141 25082 25111 D cgeo    : [RxCachedThreadScheduler-2] Database clean: removing non-existing logs from logimages
03-17 17:24:05.146 25082 25082 D cgeo    : [main] LocationProvider: requesting most precise locations
03-17 17:24:05.161 25082 25111 D cgeo    : [RxCachedThreadScheduler-2] Database clean: finished
03-17 17:24:05.473 25082 25106 D cgeo    : [OkHttp] 200 (617 ms) GET https://status.cgeo.org/api/status.json?version_code=30170317&version_name=2017.03.17-NB1-07f2df9&locale=de_DE&installer=com.google.android.packageinstaller (http/1.1)
03-17 17:24:06.066 25082 25115 D cgeo    : [OkHttp] 200 (1146 ms) GET https://www.geocaching.com/account/login (http/1.1) (=> https://www.geocaching.com/play/search)
03-17 17:24:06.569 25082 25114 I cgeo    : [network--2] Already logged in Geocaching.com as Lineflyer (PREMIUM)
03-17 17:24:06.924 25082 25115 D cgeo    : [OkHttp] GET https://www.geocaching.com/my/souvenirs.aspx
03-17 17:24:09.342 25082 25115 D cgeo    : [OkHttp] 200 (2418 ms) GET https://www.geocaching.com/my/souvenirs.aspx (http/1.1)
03-17 17:24:10.413 25082 25115 D cgeo    : [OkHttp] POST https://www.geocaching.com/my/souvenirs.aspx
03-17 17:24:12.792 25082 25115 D cgeo    : [OkHttp] 200 (2378 ms) POST https://www.geocaching.com/my/souvenirs.aspx (http/1.1)
03-17 17:24:12.792 25082 25114 I cgeo    : [network--2] changed language on geocaching.com to English
03-17 17:24:12.794 25082 25115 D cgeo    : [OkHttp] GET https://www.geocaching.com/account/login
03-17 17:24:13.893 25082 25115 D cgeo    : [OkHttp] 200 (1099 ms) GET https://www.geocaching.com/account/login (http/1.1) (=> https://www.geocaching.com/play/search)
03-17 17:24:14.373 25082 25114 I cgeo    : [network--2] Already logged in Geocaching.com as Lineflyer (PREMIUM)
03-17 17:24:14.479 25082 25114 I cgeo    : [network--2] Geocaching.com language already set to English
03-17 17:24:14.481 25082 25115 D cgeo    : [OkHttp] GET https://www.geocaching.com/email/
03-17 17:24:14.685 25082 25115 D cgeo    : [OkHttp] 200 (204 ms) GET https://www.geocaching.com/email/ (http/1.1)
03-17 17:24:15.863 25082 25106 D cgeo    : [OkHttp] GET https://www.geocaching.com/account/settings/membership
03-17 17:24:15.865 25082 25115 D cgeo    : [OkHttp] GET https://www.geocaching.com/account/settings/homelocation
03-17 17:24:15.867 25082 25303 D cgeo    : [OkHttp] GET https://www.geocaching.com/account/settings/preferences
03-17 17:24:16.187 25082 25106 D cgeo    : [OkHttp] 200 (323 ms) GET https://www.geocaching.com/account/settings/membership (http/1.1)
03-17 17:24:16.282 25082 25106 D cgeo    : [OkHttp] Setting member status to PREMIUM
03-17 17:24:17.143 25082 25115 D cgeo    : [OkHttp] 200 (1277 ms) GET https://www.geocaching.com/account/settings/homelocation (http/1.1)
03-17 17:24:17.519 25082 25303 D cgeo    : [OkHttp] 200 (1652 ms) GET https://www.geocaching.com/account/settings/preferences (http/1.1)
03-17 17:24:17.999 25082 25114 D cgeo    : [network--2] Setting GC custom date to dd MMM yy
```

**Comment by samueltardieu** [comment]
@Lineflyer I'm building -NB2 with another method to switch to English which does not require using the souvenirs page.

**Comment by Lineflyer** [comment]
@samueltardieu Looks good (and faster). Will test completely tonight and tomorrow.

**Comment by Lineflyer** [comment]
I tested several procedures once again with -NB2 incl.
- Switch to english after changed on website
- Change of userdata in c:geo
- Login with incorrect user data

All seems good.

@samueltardieu 
Please merge to `release` and `cgeo-legacy`.
As changing the login process is crucial I will release another RC as beta version afterwards.

**Comment by samueltardieu** [comment]
@Lineflyer I'll let you close it when you confirm that every branch is ok.

**Comment by Lineflyer** [comment]
@samueltardieu 
Regarding `cgeo-legacy` we might need to do something before being able to build it via CI.
See comment from @kumy on IRC:
[12:56] <kumy> OneEyed_, is this ok to merge cgeo-legacy-ci into cgeo-legacy. Then also apply the fix into release/master?
[12:56] <kumy> https://github.com/cgeo/cgeo/issues/6334#issuecomment-286581253
[12:58] <kumy> Lineflyer, be carefull, if nothing is done, "next" builds of legacy will still be build on cgeo-legacy-ci branch fork, last fix for  #6404 will not be included.
[

**Comment by Lineflyer** [comment]
Working well on all branches. Thanks @samueltardieu and @kumy
